File: Array/0217-Duplicate/Solution.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        num_set = set()
        for num in nums:
            if num in num_set:
                return True
            num_set.add(num)
        return False

"""
Space Complexity: O(n)
Time Complexity: O(n)
"""

File: Array/0217-Duplicate/Solution.java
import java.util.HashSet;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> numSet = new HashSet<>();
        for (int num : nums) {
            if (numSet.contains(num)) {
                return true;
            }
            numSet.add(num);
        }
        return false;
    }
}

/*  
Space Complexity: O(n)
Time Complexity: O(n)
*/